tests/build_/test_build_request.py

```diff
@@ -1267,14 +1267,34 @@ def test_render_prod_with_pulp_no_auth(self):
         with pytest.raises(OsbsValidationException):
             build_request.render()
 
+    def test_render_prod_with_empty_triggers(self, tmpdir):
+
+        self.create_image_change_trigger_json(str(tmpdir), custom_triggers=[])
+        build_request = BuildRequest(str(tmpdir))
+        kwargs = get_sample_prod_params()
+        build_request.set_params(**kwargs)
+        build_request.render()
+
     @staticmethod
-    def create_image_change_trigger_json(outdir):
+    def create_image_change_trigger_json(outdir, custom_triggers=None):
         """
         Create JSON templates with an image change trigger added.
 
         :param outdir: str, path to store modified templates
         """
 
+        triggers = custom_triggers if custom_triggers is not None else [
```

Comment:
Nitpick: its a bit tough to read this test. Instead of setting it in the test function we should be passing it using `parametrize`:
```
    @pytest.mark.parametrize(('triggers'), (
        None,
        <here goes that long dict>
    ))
    def create_image_change_trigger_json(outdir, triggers):
```

This way the test would run twice with two different params, which could later on be extended.